code/game/objects/items/tools/medical_tools.dm

```diff
@@ -0,0 +1,246 @@
+	if(brute)
+		brute = FALSE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 5)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 15)  //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] to burn mode.</span>",
+		"<span class='notice'>You switch [src] lever to burn mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		update_kitprinter()
+		return
+
+	if(!brute)
+		brute = TRUE
+		playsound(loc, 'sound/machines/computer_typing1.ogg', 15)
+		playsound(loc, 'sound/machines/terminal_on.ogg', 5 , 1) //maybe placehold. maybe it will be left as this. but sure its better than a bland click
+		if(!silent && user)
+			user.visible_message("<span class='notice'>[user] turns [src] to brute mode.</span>",
+		"<span class='notice'>You switch [src] lever to brute mode. <b>Gel gauge reads: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		update_kitprinter()
+		return
+
+/obj/item/tool/geltool/gadget/kitprinter/attack(mob/living/carbon/M as mob, mob/user as mob)
+
+	if(!cell || cell.gelquantity < gelefficiency)
+		to_chat(user, "<span class='warning'>The [src] is out of gel.</span>")
+		return
+
+	if (!ishuman(M))
+		return
+
+	var/mob/living/carbon/human/H = M
+	var/datum/limb/affecting = user.client.prefs.toggles_gameplay & RADIAL_MEDICAL ? radial_medical(H, user) : H.get_limb(user.zone_selected)
+
+	if(!affecting)
+		return TRUE
+
+	if(affecting.surgery_open_stage != 0)
+		to_chat(user, "<span class='warning'>Close the incision before using a [src].</span>")
+		return
+
+	var/bandaged = affecting.bandage()
+	var/disinfected = affecting.disinfect()
+	var/heal_amt = heal_rate
+	var/datum/wound/W = affecting.wounds
+
+	if(brute)
+		if(user.skills.getRating("medical") < SKILL_MEDICAL_PRACTICED) //untrained marines have a hard time using it
+			to_chat(user, "<span class='warning'>You start fumbling with [src].</span>")
+			if(!do_mob(user, M, 6 SECONDS, BUSY_ICON_UNSKILLED, BUSY_ICON_MEDICAL)) //yes 2x the advanced pack
+				return
+
+		if(!(bandaged || disinfected))
+			to_chat(user, "<span class='warning'>The wounds on [M]'s [affecting.display_name] have already been treated.</span>")
+			return
+
+		user.visible_message("<span class='notice'>[user] applies the printkit on [W.desc] on [M]'s [affecting.display_name].</span>",
+		"<span class='notice'>You pressure the printer against [W.desc] on [M]'s [affecting.display_name].</span>")
+		if(bandaged)
+			affecting.heal_limb_damage(heal_amt, updating_health = TRUE)
+		gel_use()
+		return
+
+	if(!brute)
+		if(user.skills.getRating("medical") < SKILL_MEDICAL_PRACTICED) //untrained marines have a hard time using it
+			to_chat(user, "<span class='warning'>You start fumbling with [src].</span>")
+			if(!do_mob(user, M, 6 SECONDS, BUSY_ICON_UNSKILLED, BUSY_ICON_MEDICAL)) //yes 2x the advanced pack
+				return
+
+		if(!(bandaged || disinfected))
```

Comment:
the logic seems wrong here?

Reply:
if its either bandaged or disinfected
then the wound was already treated...

i dont disagree but i dont see where is my mistake still